microsoft/PowerToys
# [PT Run] Web search should use user's default browser #16549

## Summary of the Pull Request

**What is this about:**
Before this PR PT Run used Application or Default Icon location specified in the registry to find default browser program location. If PT Run was unable to detect program location it used Microsoft Edge.
Also for Opera browser implementation checked internal Opera settings and started a search engine specified with constructed address.

**What is included in the PR:** 
Default Browser Info detection logic updated to use shell command for default browser.
Parameters specified in command are used and search terms are working for Chrome, Firefox, Microsoft Edge and Opera.
If we're unable to find default browser we're using Microsoft Edge.

**How does someone test / validate:** 
Start Power Toys
Go to Power Toys Run
Type ?? to start Web Search plugin
Check your default browser is mentioned in the suggestion area
Close Power Toys
Change your default browser
Repeat previous steps and check that a new default browser shown in the suggestion area

## Quality Checklist

- [x] **Linked issue:** #16549 

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
